src/Ghcid.hs

```diff
@@ -195,15 +195,22 @@ runGhcid waiter restart command outputfiles test size titles output = do
             outputFill (Just (loadedCount, messages)) ["Running test..." | isJust test]
             forM_ outputfiles $ \file ->
                 writeFile file $ unlines $ map snd $ prettyOutput loadedCount $ filter isMessage messages
-            whenJust test $ \test -> do
-                res <- exec ghci test
-                outputFill (Just (loadedCount, messages)) $ fromMaybe res $ stripSuffix ["*** Exception: ExitSuccess"] res
-                updateTitle ""
+            whenJust test $ \t -> do
+                whenLoud $ outStrLn $ "%TESTING: " ++ t
+                forkIO $ do
+                    res <- exec ghci t
+                    whenLoud $ outStrLn "%TESTING: Completed"
+                    outputFill (Just (loadedCount, messages)) $ fromMaybe res $ stripSuffix ["*** Exception: ExitSuccess"] res
+                    -- setSGR [Reset]
+                    updateTitle ""
+                return ()
 
             when (null wait) $ do
```

Comment:
If we can move this check before executing test command we can be sure we can drop the nested interrupt